syndesis/plugins/org.fusesource.ide.syndesis.extension.ui/src/org/fusesource/ide/syndesis/extensions/ui/SyndesisExtensionProjectNature.java

```diff
@@ -0,0 +1,86 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.fusesource.ide.syndesis.extensions.ui;
+
+import org.eclipse.core.resources.ICommand;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IProjectDescription;
+import org.eclipse.core.resources.IProjectNature;
+import org.eclipse.core.runtime.CoreException;
+import org.fusesource.ide.project.RiderProjectBuilder;
+
+/**
+ * @author lheinema
+ */
+public class SyndesisExtensionProjectNature implements IProjectNature {
+
+	public static final String NATURE_ID = "org.fusesource.ide.syndesis.extension";
+	
+	private IProject project;
+	
+	/* (non-Javadoc)
+	 * @see org.eclipse.core.resources.IProjectNature#configure()
+	 */
+	@Override
+	public void configure() throws CoreException {
+		IProjectDescription desc = project.getDescription();
+		ICommand[] commands = desc.getBuildSpec();
+
+		for (int i = 0; i < commands.length; ++i) {
+			if (commands[i].getBuilderName().equals(RiderProjectBuilder.BUILDER_ID)) {
+				return;
+			}
+		}
+
+		ICommand[] newCommands = new ICommand[commands.length + 1];
+		System.arraycopy(commands, 0, newCommands, 0, commands.length);
+		ICommand command = desc.newCommand();
+		command.setBuilderName(RiderProjectBuilder.BUILDER_ID);
```

Comment:
why do we have a RiderProjectBuilder on the Syndesis project?

Reply:
its actually just a java builder and it builds the same way as a fuse integration projects which also uses this.

Reply:
Why do we need this builder?

Reply:
I am rather unsure. Guess we can even remove it.

Reply:
ok, removed nature and facet